integration/compose/src/main/java/com/bumptech/glide/integration/compose/GlideModifier.kt

```diff
@@ -0,0 +1,252 @@
+package com.bumptech.glide.integration.compose
+
+import android.graphics.PointF
+import android.graphics.drawable.Drawable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.graphics.asAndroidColorFilter
+import androidx.compose.ui.graphics.drawscope.ContentDrawScope
+import androidx.compose.ui.graphics.drawscope.translate
+import androidx.compose.ui.graphics.nativeCanvas
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.layout.Measurable
+import androidx.compose.ui.layout.MeasureResult
+import androidx.compose.ui.layout.MeasureScope
+import androidx.compose.ui.layout.times
+import androidx.compose.ui.node.DrawModifierNode
+import androidx.compose.ui.node.LayoutModifierNode
+import androidx.compose.ui.node.ModifierNodeElement
+import androidx.compose.ui.node.SemanticsModifierNode
+import androidx.compose.ui.node.invalidateDraw
+import androidx.compose.ui.semantics.SemanticsPropertyKey
+import androidx.compose.ui.semantics.SemanticsPropertyReceiver
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.IntSize
+import com.bumptech.glide.RequestBuilder
+import com.bumptech.glide.integration.ktx.AsyncGlideSize
+import com.bumptech.glide.integration.ktx.ExperimentGlideFlows
+import com.bumptech.glide.integration.ktx.ImmediateGlideSize
+import com.bumptech.glide.integration.ktx.InternalGlideApi
+import com.bumptech.glide.integration.ktx.Placeholder
+import com.bumptech.glide.integration.ktx.ResolvableGlideSize
+import com.bumptech.glide.integration.ktx.Resource
+import com.bumptech.glide.integration.ktx.Status
+import com.bumptech.glide.integration.ktx.flowResolvable
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.plus
+import kotlin.math.roundToInt
+
+@ExperimentalGlideComposeApi
+@OptIn(InternalGlideApi::class)
+internal fun Modifier.glideNode(
+  requestBuilder: RequestBuilder<Drawable>,
+  size: ResolvableGlideSize,
+  contentScale: ContentScale,
+  alignment: Alignment,
+  colorFilter: ColorFilter?,
+): Modifier =
+  this then GlideNodeElement(requestBuilder, size, contentScale, alignment, colorFilter)
+
+@ExperimentalGlideComposeApi
+@OptIn(InternalGlideApi::class)
+internal data class GlideNodeElement constructor(
+  private val requestBuilder: RequestBuilder<Drawable>,
+  private val size: ResolvableGlideSize,
+  private val contentScale: ContentScale,
+  private val alignment: Alignment,
+  private val colorFilter: ColorFilter?,
```

Comment:
you should make sure that all of the classes you are using here have sensible equals functions. The GlideNodeElement will be reallocated on recompositions and if nothing for this image has changed you want this element to compare equals with the one created in the previous composition. It is only when these elements are not equal that `update` will get called

Reply:
Done, thanks for pointing this out!